### PR TITLE
Avoid throwing and catching InvalidGroupException over and over when consumer don't use any consumer group id

### PR DESCRIPTION
**Pull request by strokyl**
The zio-kafka consumer was trying to always get consumer group metada even if the consumer wasn't using any id.
This was fine because it was doing into a `Try` but this wasn't good for performance!

### CODE REVIEW

**Comment by guizmaii** [comment]
@strokyl Isn't your title+comment in this PR weird compared to what you're working on? 🤔

**Comment by strokyl** [comment]
> @strokyl Isn't your title+comment in this PR weird compared to what you're working on? thinking

I don't see what you are talking about? ;)
No seriously I thought I did a commit but didn't so I amend previous comment on cdk-master therefore the weird message and etc.. etc..
All of that is fixed now

File: src/main/scala/zio/kafka/consumer/ConsumerSettings.scala [hunk 1]
```diff
@@ -37,6 +37,9 @@ final case class ConsumerSettings(
   def withGroupId(groupId: String): ConsumerSettings =
     withProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId)
 
+  def hasGroupId: Boolean =
+    properties.contains(ConsumerConfig.GROUP_ID_CONFIG)
+
```
Comment (azhur) [inline src/main/scala/zio/kafka/consumer/ConsumerSettings.scala:42]: 🟡 i would not expose this method to the lib clients, I would inline where it is called.

File: src/main/scala/zio/kafka/consumer/internal/Runloop.scala [hunk 2]
```diff
@@ -189,9 +188,21 @@ private[consumer] final class Runloop(
             reqRecs.toArray[ByteArrayConsumerRecord](Array.ofDim[ByteArrayConsumerRecord](reqRecs.size))
           )
 
-        fulfillAction = fulfillAction *> req.cont.succeed(
-          concatenatedChunk.map(CommittableRecord(_, commit(_), Try(consumer.consumer.groupMetadata()).toOption))
-        )
+        fulfillAction = fulfillAction *> req.cont.succeed(concatenatedChunk.map { record =>
+          CommittableRecord(
+            record = record,
+            commitHandle = commit(_),
```
Comment (guizmaii) [inline src/main/scala/zio/kafka/consumer/internal/Runloop.scala]:
```suggestion
            commitHandle = commit,
```

**Review by guizmaii — APPROVED** [review]
🎖

File: src/main/scala/zio/kafka/consumer/ConsumerSettings.scala (hunk 1 above)
Comment (strokyl) [reply src/main/scala/zio/kafka/consumer/ConsumerSettings.scala:42]: I made it private

**Review by azhur — APPROVED** [review]

**Pull request merged**